docs/js/general.js

```diff
@@ -137,6 +138,30 @@ function set_genre_tags(user_tags) {
     $('#genre_tags_opt').trigger("change"); // Update selection
 }
 
+function set_answer_setting_table_to(and_or) {
+    const table = document.getElementById("answersetting");
+
+    let display;
+    let invisible;
+    if (and_or === "and") {
+        display = ["visible", "none"];
+        invisible = [...table.getElementsByClassName("solcheck_or")];
+    } else if (and_or === "or") {
+        display = ["none", "visible"];
+        invisible = [...table.getElementsByClassName("solcheck")];
+    } else {
+        return;
+    }
+    // Ensure there are no invisible checked boxes
+    invisible.forEach((elem) => {elem.checked = false});

```

Comment:
I get the idea where this is coming from, 
But this is causing an issue because every time you switch between All constraints and Any of the constraints, everything gets reset, I don't think that would be ideal as users may frequently be switching just to check out, and they will lose their selection.
Remember, a priority is already established for "ALL" in the solution check. Not that this approach is wrong, but I think leaving to default behavior where no one has any issues so far is always a good thing.
I am disabling this line for now.

Reply:
But that's the issue, ALL has priority. So if you select options in ALL, realize you wanted ANY, and switch the options in ANY, the options in ALL will still have priority when you don't realize it.

I agree it's inconvenient for switching between the two, but preventing user error, especially errors with an invisible cause, is much more important imo.

Reply:
I understand and agree with you. Let's see, I will leave it as is for now, and if there is some feedback over it, then will uncomment it in the next update.